File: eBiblioteka.Servisi/Database/Knjiga.cs
using System;
using System.Collections.Generic;

namespace eBiblioteka.Servisi.Database;

public partial class Knjiga
{
    public int KnjigaId { get; set; }

    public string Naziv { get; set; } = null!;

    public string? KratkiOpis { get; set; }

    public int? GodinaIzdanja { get; set; }

    public byte[]? Slika { get; set; }

    public int? ZanrId { get; set; }

    public bool? IsDeleted { get; set; }

    public DateTime? VrijemeBrisanja { get; set; }

    public int Kolicina { get; set; }

    public bool? Dostupna { get; set; }

    public bool? KnjigaDana { get; set; }

    public bool? Preporuceno { get; set; }

    public virtual ICollection<KnjigaAutor> KnjigaAutors { get; set; } = new List<KnjigaAutor>();

    public virtual ICollection<KorisnikIzabranaKnjiga> KorisnikIzabranaKnjigas { get; set; } = new List<KorisnikIzabranaKnjiga>();

    public virtual ICollection<Recenzija> Recenzijas { get; set; } = new List<Recenzija>();

    public virtual ICollection<Rezervacija> Rezervacijas { get; set; } = new List<Rezervacija>();

    public virtual Zanr? Zanr { get; set; }
}

File: eBibliotekaUI/ebiblioteka_web/lib/models/knjiga.dart
import 'package:json_annotation/json_annotation.dart';

part 'knjiga.g.dart';

@JsonSerializable()
class Knjiga {
  int? knjigaId;
  String? naziv;
  String? kratkiOpis;
  int? godinaIzdanja;
  String? slika;
  int? zanrId;
  bool? isDeleted;
  DateTime? vrijemeBrisanja;
  int? kolicina;
  bool? dostupna;
  bool? knjigaDana;
  bool? preporuceno;
  List<int>? autoriIds;

  Knjiga({
    this.knjigaId,
    this.naziv,
    this.autoriIds,
  });

  factory Knjiga.fromJson(Map<String, dynamic> json) => _$KnjigaFromJson(json);

  Map<String, dynamic> toJson() => _$KnjigaToJson(this);
}
